Lib/profiling/sampling/cli.py

```diff
@@ -0,0 +1,692 @@
+    """
+    collector_class = COLLECTOR_MAP.get(format_type)
+    if collector_class is None:
+        raise ValueError(f"Unknown format: {format_type}")
+
+    # Gecko format never skips idle (it needs both GIL and CPU data)
+    if format_type == "gecko":
+        skip_idle = False
+
+    return collector_class(interval, skip_idle=skip_idle)
+
+
+def _generate_output_filename(format_type, pid):
+    """Generate output filename based on format and PID.
+
+    Args:
+        format_type: The output format
+        pid: Process ID
+
+    Returns:
+        Generated filename
+    """
+    extension = FORMAT_EXTENSIONS.get(format_type, "txt")
+    return f"{format_type}.{pid}.{extension}"
+
+
+def _handle_output(collector, args, pid, mode):
+    """Handle output for the collector based on format and arguments.
+
+    Args:
+        collector: The collector instance with profiling data
+        args: Parsed command-line arguments
+        pid: Process ID (for generating filenames)
+        mode: Profiling mode used
+    """
+    if args.format == "pstats":
+        if args.outfile:
+            collector.export(args.outfile)
+        else:
+            # Print to stdout
+            sort_mode = _sort_to_mode(args.sort)
+            collector.print_stats(
+                sort_mode, args.limit, not args.no_summary, mode
+            )
+    else:
+        # Export to file
+        filename = args.outfile or _generate_output_filename(args.format, pid)
+        collector.export(filename)
+
+
+def _validate_args(args, parser):
+    """Validate format-specific options and live mode requirements.
+
+    Args:
+        args: Parsed command-line arguments
+        parser: ArgumentParser instance for error reporting
+    """
+    # Check if live mode is available
+    if args.command == "live" and LiveStatsCollector is None:
+        parser.error(
+            "Live mode requires the curses module, which is not available."
+        )
+
+    # Only validate format options for run/attach commands (live doesn't have format option)
+    if args.command not in ("run", "attach"):
+        return
+
+    # Validate gecko mode doesn't use non-wall mode
+    if args.format == "gecko" and args.mode != "wall":
+        parser.error(
+            "--mode option is incompatible with --gecko. "
+            "Gecko format automatically includes both GIL-holding and CPU status analysis."
+        )
+
+    # Validate pstats-specific options are only used with pstats format
+    if args.format != "pstats":
+        issues = []
+        if args.sort != "nsamples":
+            issues.append("--sort")
+        if args.limit != 15:
+            issues.append("--limit")
+        if args.no_summary:
+            issues.append("--no-summary")
+
+        if issues:
+            format_flag = f"--{args.format}"
+            parser.error(
+                f"Options {', '.join(issues)} are only valid with --pstats, not {format_flag}"
+            )
+
+
```

Comment:
If we add something like this:
```suggestion
class CustomFormatter(
    argparse.ArgumentDefaultsHelpFormatter,
    argparse.RawDescriptionHelpFormatter,
):
    pass


```